### DISCUSSION TITLE
Is there a way to access the user id of the user calling an edge function?

### DISCUSSION THREAD

**Discussion by ryankaplan** [supabase/supabase#18803] [Questions]
Edge functions are authenticated by default (logged out users can't call them) but I don't see an obvious API for finding the user id of the user calling an edge function. Is that available somewhere?

**Comment by GaryAustin1** [comment]
https://supabase.com/docs/guides/functions/auth
The entire user jwt should be passed.

**Comment by joeldomke** [comment]
As long as JWT verification is not disabled for the EF, can we do the following and avoid the network call?

```ts
const authHeader = req.headers.get('Authorization')!;
const token = authHeader.replace('Bearer ', '');

const payload = JSON.parse(atob(token.split('.')[1]));
const userId = payload.sub;
```

Actually, how does the EF verify the JWT? The JWT is valid for 1 hour. What happens when the auth user is deleted in that hour? Can the user still call the EF as an authenticated user during that hour? In this case the network call would make sense to ensure that the user still exists.
